packages/expo-updates/ios/EXUpdates/Logging/UpdatesLogEntry.swift

```diff
@@ -0,0 +1,79 @@
+// Copyright 2022-present 650 Industries. All rights reserved.
+
+import Foundation
+
+/**
+ Schema for the fields in expo-updates log message JSON strings
+ */
+public struct UpdatesLogEntry: Codable {
+  var timestamp: UInt // seconds since 1/1/1970 UTC
+  var message: String
+  var code: String // One of the UpdatesErrorCode string values
+  var level: String // One of the ExpoModulesCore.LogType string values
+  var updateId: String? // EAS update ID, if any
+  var assetId: String? // EAS asset ID, if any
+  var stacktrace: [String]? // Stacktrace (for error and fatal logs)
+
+  /**
+   Returns a JSON string representation from this UpdatesLogEntry object
+   */
+  public func asString() -> String? {
+    do {
+      let jsonEncoder = JSONEncoder()
+      let jsonData = try jsonEncoder.encode(self)
+      return String(data: jsonData, encoding: .utf8)
+    } catch {
+      return nil
+    }
+  }
+
+  /**
+   Returns a Dictionary representation from this UpdatesLogEntry object
+   */
+  public func asDict() -> [String: Any] {
+    var result = [String: Any]()
+    result["timestamp"] = timestamp
+    result["message"] = message
+    result["code"] = code
+    result["level"] = level
+    if (updateId != nil) {
+      result["updateId"] = updateId
+    }
+    if (assetId != nil) {
+      result["assetId"] = assetId
+    }
+    if stacktrace != nil {
+      let nsstack = NSMutableArray()
+      for s in stacktrace! {
```

Comment:
I think I already mentioned this somewhere — never use force unwrapping `!` except in tests and some edge-cases where it makes sense and we're sure it won't crash. Here the situation is simple, just do
```
if let stacktrace = stacktrace {
```
instead of
```
if stacktrace != nil {
```
to safely unwrap the variable

Reply:
That is an interesting syntax

Reply:
Fixed